centreon/packages/ui/src/components/Menu/Button/MenuButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { ReactElement, ReactNode } from 'react';
+
+import {
+  ArrowDropDown as ArrowDropDownIcon,
+  Menu as MenuIcon
+} from '@mui/icons-material';
+
+import { Button, ButtonProps } from '../../Button';
+import { AriaLabelingAttributes } from '../../../@types/aria-attributes';
+import { DataTestAttributes } from '../../../@types/data-attributes';
+import { useMenu } from '../useMenu';
+
+import { useStyles } from './MenuButton.styles';
+
+type MenuButtonProps = {
+  children?: ReactNode;
+  hasArrow?: boolean;
+  isOpen?: boolean;
+  onClick?: (isOpen: boolean) => void;
+} & Pick<ButtonProps, 'disabled' | 'size' | 'variant'> &
+  AriaLabelingAttributes &
+  DataTestAttributes;
+
+const MenuButton = ({
+  children,
+  onClick,
+  hasArrow = true,
+  size = 'small',
+  variant = 'ghost',
+  ...attr
+}: MenuButtonProps): ReactElement => {
+  const { classes } = useStyles();
+
+  const { isMenuOpen, setIsMenuOpen, setAnchorEl, onOpen } = useMenu();
+
+  const onToggle = (e): void => {
+    setAnchorEl(e.currentTarget);
+
+    setIsMenuOpen(!isMenuOpen);
```

Comment:
is this jotai specific ? what's the benefit ?

because next line it's used in the callback also

```
    setIsMenuOpen(!isMenuOpen);
    onClick?.(!isMenuOpen);
```

Reply:
It's copied from a React feature: https://react.dev/reference/react/useCallback#updating-state-from-a-memoized-callback

Reply:
And the onClick does not need that because it's not memoized

Reply:
this would be the case if `onToggle` is actually memoized (which is not the case here)

```
const onToggle = useCallback((e): void => {
    setIsMenuOpen(!isMenuOpen);
  }, [isMenuOpen]);
```
vs.
```
const onToggle = useCallback((e): void => {
    setIsMenuOpen((currentIsMenuOpen) => !currentIsMenuOpen);
  }, []); // ✅ No need for the isMenuOpen dependency
```

...only, as `onClick` needs the value, it would be required as a dependency


that said, will memoize the `onToggle`